crates/flux-desugar/src/lib.rs

```diff
@@ -105,8 +103,18 @@ pub fn desugar<'genv>(
                         fhir::Node::Item(genv.alloc(cx.desugar_impl(owner_id, impl_)?)),
                     );
                 }
+                // hir::ItemKind::OpaqueTy(_) => {
```

Comment:
I didn't follow why these were two separate things, so I merged them and made it pop out an `unsupported` to allow progress, lmk if this should be reverted!

Reply:
Do you have a minimal example that triggers this?

This shouldn't happen because an `OpaqueTy` is desugared as part of desugaring a function, e.g., if we have 
```rust
fn foo() -> impl SomeTrait
```
the item corresponding to the opaque type generated for the `impl SomeTrait` should be desugared when we desugar `foo` so I don't see how one could be querying the `DefId` of an opaque type if the function hasn't been desugared yet.